src/main/java/seedu/address/logic/commands/UntagCommand.java

```diff
@@ -54,35 +47,38 @@ public UntagCommand(ContactIndex index, Set<ModuleTag> modulesToRemove) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        Person personToEdit = getPersonToEdit(model);
+
+        personToEdit.removeModuleTags(moduleTags);
+
+        if (personToEdit instanceof User) {
+            return setUserCommonModuleTags(model, (User) personToEdit);
+        }
 
+        return setPersonCommonModuleTags(model, personToEdit);
```

Comment:
@kennycjy Take note I put this here so that not all of your code is removed. You can clean this up for both Tag and Untag command.